### PR TITLE
BLEN-32: Move some USD Hydra options to plugin preferences

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Move some USD Hydra options to plugin preferences.
Added URL buttons to the addon preferences panel to look more like blender style.
A convenient tool for fast debugging on different platforms.

### EFFECT OF CHANGE
Added debug tools to the addon preferences panel.
Added URL buttons to the addon preferences panel.

### TECHNICAL STEPS
Added class  HDUSD_ADDON_PT_preferences.
Added tmp_dir, dev_tools, debug_log properties.
Changed operator hdusd.op_open_web_page to wm.url_open for main_web, community, downloads URLs
Removed class HDUSD_OP_open_web_page.
Added keys 'tracker_url', 'doc_url', 'community', 'downloads', 'main_web' to bl_info. 
Code improvements.

### CODE REVIEW

**Comment by VascoPi** [comment]
image

File: src/hdusd/__init__.py [hunk 1]
```diff
@@ -21,15 +21,80 @@
     "location": "Info header, render engine menu",
     "description": "USD Hydra rendering plugin for Blender",
     "warning": "",
-    "tracker_url": "",
-    "doc_url": "",
-    "category": "Render"
+    "tracker_url": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/issues",
+    "doc_url": "https://radeon-pro.github.io/RadeonProRenderDocs/en/usd_hydra/about.html",
```
Comment (bnagirniak) [inline src/hdusd/__init__.py:25]: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon#readme

File: src/hdusd/__init__.py [hunk 2]
```diff
@@ -21,15 +21,80 @@
...
+        update=update_temp_dir,
+    )
+    dev_tools: BoolProperty(
+        name="Developer Tools",
+        description="Enable developer tools",
+        default=config.show_dev_settings,
+        update=update_dev_tools,
+    )
+    debug_log: BoolProperty(
+        name="Debug",
```
Comment (bnagirniak) [inline src/hdusd/__init__.py:81]: Debug Logging

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Suggested to change links. Do we need Help & About panel?
Discuss with @bsavery

**Comment by VascoPi** [comment]
@Brian
Could you please look if URLs are correct for BlenderUSDAddon.
  tracker_url: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/issues,
  doc_url: https://radeon-pro.github.io/RadeonProRenderDocs/en/usd_hydra/about.html,
  community: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/discussions,
  downloads: https://www.amd.com/en/technologies/radeon-prorender-downloads",
  main_web: https://www.amd.com/en/technologies/radeon-prorender,

**Comment by bsavery** [comment]
> @bsavery Could you please look if URLs are correct for BlenderUSDAddon. tracker_url: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/issues, doc_url: https://radeon-pro.github.io/RadeonProRenderDocs/en/usd_hydra/about.html, community: https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/discussions, downloads: https://www.amd.com/en/technologies/radeon-prorender-downloads", main_web: https://www.amd.com/en/technologies/radeon-prorender,
> 
> Also, there're proposals in recent comments.

downloads should point at releases on GitHub repo

File: src/hdusd/__init__.py [hunk 3]
```diff
@@ -21,15 +21,80 @@
...
+        default=str(temp_dir()),
+        update=update_temp_dir,
+    )
+    dev_tools: BoolProperty(
+        name="Developer Tools",
+        description="Enable developer tools",
+        default=config.show_dev_settings,
+        update=update_dev_tools,
+    )
+    debug_log: BoolProperty(
```
Comment (DagerD) [inline src/hdusd/__init__.py:80]: Please add other levels.

**Review by DagerD — CHANGES_REQUESTED** [review]
see comment

**Comment by VascoPi** [comment]
image

**Review by DagerD — APPROVED** [review]

File: src/hdusd/__init__.py [hunk 4]
```diff
@@ -21,15 +21,86 @@
...
+        bpy.context.preferences.addons[__name__].preferences['tmp_dir'] = str(temp_dir())
+        log.info(f"Current temp directory is changed to {bpy.context.preferences.addons[__name__].preferences.tmp_dir}")
+
+    def update_dev_tools(self, context):
+        config.show_dev_settings = self.dev_tools
+        log.info(f"Developer settings is {'enabled' if self.dev_tools else 'disabled'}")
+
+    def update_log_level(self, context):
+        logging.logger.setLevel(self.log_level)
+        log.critical(f"Log level is set to {self.log_level}")
```
Comment (bnagirniak) [inline src/hdusd/__init__.py:64]: Why log.critical() is here? I think log.info() should be

**Review by bnagirniak — APPROVED** [review]
Tested - works good. 
Just one notation about using log.critical().

File: src/hdusd/__init__.py (hunk 4 above)
Comment (VascoPi) [reply src/hdusd/__init__.py:64]: The purpose is to output update_log_level to a console regardless of the log level.

**Pull request merged**